src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -0,0 +1,337 @@
+    modify = False
+
+    for rdoc in rules:
+        # query Rucio for that rule id and fetch its state (e.g.: afd122143kjmdskj)
+        rse = rdoc['rse_expression']
+        rid = rdoc['id']
+        state = rdoc['state']
+        account = rucioClient.whoami()
+
+        # rucio state have the following values
+        # https://github.com/rucio/rucio/blob/master/lib/rucio/db/sqla/constants.py
+        # the states are: OK, REPLICATING, STUCK, SUSPENDED, WAITING_APPROVAL, INJECT
+        if state == 'OK':
+            # log that the rule has been satisfied and add that RSE to the currentRSEs (unique)
+            msg = f"Container {pname} under the rule ID {rid} targeting RSE {rse} in rucio account {account} is completely available"
+            logger.info(msg)
+            rses = doc['currentRSEs']
+            if rse not in rses:
+                rses.append(rse)
+                doc['currentRSEs'] = list(set(rses))
+                modify = True
+        else:
+            # calculate the rule completion based on the 3 locks_* field
+            sumOfLocks = rdoc['locks_ok_cnt'] + rdoc['locks_replicating_cnt'] + rdoc['locks_stuck_cnt']
+            completion = rdoc['locks_ok_cnt'] / sumOfLocks
+            msg = f"Container {pname} under the rule ID {rid} targeting RSE {rse} in rucio account {account} has a fraction completion of {completion}"
+            logger.info(msg)
+            modify = True
+
+    # persist an up-to-date version of the pileup data structure in MongoDB
+    if modify:
+        logger.info("monitoring task update %s", doc)
+        mgr.updatePileup(doc)
+
+
+async def inactiveTask(doc, spec):
+    """
+    Perform single inactive task over provided MSPileup document
+    :param mgr: MSPileupData manager
+    :param doc: MSPileup document
+    :param spec: task spec dict
+    """
+    mgr = spec['manager']
+    rucioClient = spec['rucioClient']
+    logger = spec.get('logger', getMSLogger(False))
+    pname = doc['pileupName']
+    inputArgs = {"scope": "cms", "name": pname}
+    ruleIds = [r for r in rucioClient.list_replication_rules(inputArgs)]
+
+    for rdoc in list(ruleIds):
+        # make a Rucio call to delete that rule id
+        rid = rdoc['id']
+        rucioClient.delete_replication_rule(rid)
+
+        # remove the rule id from rulesList (if any) and remove the RSE name from currentRSEs (if any)
+        rse = rdoc['rse_expression']
+        rses = doc['currentRSEs']
+        if rid in ruleIds:
+            ruleIds.remove(rid)
+        if rse in rses:
+            rses.remove(rse)
+            doc['currentRSEs'] = rses
+
+    # make a log record if the DID + Rucio account tuple does not have any existent rules
+    if len(ruleIds):
+        account = rucioClient.whoami()
+        msg = f"Container: {pname} for Rucio account {account} does not have any existing rules"
+        logger.info(msg)
+
+        # and set rulesList and currentRSEs to an empty list
```

Comment:
I can remove it, but it will contradict your instructions which wrote the following:
```
3. make a log record if the DID + Rucio account tuple does not have any existent rules
* and set rulesList and currentRSEs to an empty list
```
So, we either need to change instruction logic or you must clarify further why it it is safe to remove this assingments.

Reply:
done